packages/core/core/src/services/document-service/internationalization.ts

```diff
@@ -37,26 +37,15 @@ const localeToLookup: Transform = (contentType, params) => {
   }
 
   if (params.locale) {
-    return assoc(['lookup', 'locale'], params.locale, params);
-  }
-
-  return params;
-};
+    if (typeof params.locale === 'string') {
+      if (params.locale === '*') {
+        return params;
+      }
 
-/**
- * Add locale lookup query to the params
- */
-const multiLocaleToLookup: Transform = (contentType, params) => {
-  if (!strapi.plugin('i18n').service('content-types').isLocalizedContentType(contentType)) {
-    return params;
-  }
-
-  if (params.locale) {
-    if (params.locale === '*') {
-      return params;
+      return assoc(['lookup', 'locale'], params.locale, params);
     }
 
-    return assoc(['lookup', 'locale'], params.locale, params);
+    return assoc(['lookup', 'locale', '$in'], params.locale, params);
```

Comment:
this might not be an array if it's not a string it might be sth else do we validate the param somewhere else ? 

the $in is not necessary by the way so I would simplify as the array in a where will be automatically managed